scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -20,23 +20,25 @@ def __init__(self, auth_encoding='latin-1'):
         for type, url in getproxies().items():
             self.proxies[type] = self._get_proxy(url, type)
 
-        if not self.proxies:
-            raise NotConfigured
-
     @classmethod
     def from_crawler(cls, crawler):
+        if not crawler.settings.getbool('HTTPPROXY_ENABLED'):
+            raise NotConfigured
         auth_encoding = crawler.settings.get('HTTPPROXY_AUTH_ENCODING')
         return cls(auth_encoding)
 
+    def _basic_auth_header(self, username, password):
+        user_pass = to_bytes(
+            '%s:%s' % (unquote(username), unquote(password)),
+            encoding=self.auth_encoding)
+        return base64.b64encode(user_pass).strip()
```

Comment:
I don't really know, what I did was move that code to a separate function to be able to reuse it. I think the `w3lib` version should work just fine here, the only thing missing would be the ability to define encoding.